### PR TITLE
update kyaml to v0.12 and cli-utils to latest

### PR DESCRIPTION
**Pull request by natasha41575**
This PR updates kpt to kyaml v0.12.0 and includes an important change referenced by https://github.com/GoogleContainerTools/kpt/issues/2478 and https://github.com/kubernetes-sigs/kustomize/issues/4024.

The following annotations are being migrated: 

  - `config.k8s.io/id` -> `internal.config.kubernetes.io/id`
  - `config.kubernetes.io/path` -> `internal.config.kubernetes.io/path`
  - `config.kubernetes.io/index` -> `internal.config.kubernetes.io/index`

For some time, we will support both annotations by adding both legacy and the new annotation when reading files. When writing files, the orchestrator consumes both and reconciles correctly by comparing the resource before and after running a function to infer which annotation has changed (if they differ). See the kyaml PR that implemented this change here: https://github.com/kubernetes-sigs/kustomize/pull/4190.

After some grace period, support for the legacy annotations will be dropped.

### CODE REVIEW

File: e2e/live/end-to-end-test.sh [hunk 1]
```diff
@@ -828,12 +828,7 @@ echo "Testing apply/prune after migrate"
 echo "kpt live apply e2e/live/testdata/migrate-case-1b"
 cp -f e2e/live/testdata/migrate-case-1a/Kptfile e2e/live/testdata/migrate-case-1b
 ${BIN_DIR}/kpt live apply e2e/live/testdata/migrate-case-1b > $OUTPUT_DIR/status
-assertContains "namespace/test-rg-namespace unchanged"
```
Comment (mengqiy) [inline e2e/live/end-to-end-test.sh:831]: Why we touch things related to `live`?

**Review by mengqiy — COMMENTED** [review]
IMO we should add some e2e tests for it. For example, we can have something like:
Prepare 2 resources in one file. Run a function (can be written in starlark) to update the new path and index annotations.
Before:
```yaml
# x.yaml
resourceFoo
---
resourceBar
```
After:
```yaml
# y.yaml
resourceBar
---
resourceFoo
```
Do the same using the legacy annotations in another test. It should produce the same result.

**Comment by natasha41575** [comment]
Adding e2e tests revealed an issue; will investigate and fix.

**Review by mengqiy — APPROVED** [review]
LGTM

**Comment by mengqiy** [comment]
@mortent Do you want to take another look given you requested changes earlier in https://github.com/GoogleContainerTools/kpt/pull/2523#pullrequestreview-779279647?

File: internal/cmdrender/executor.go [hunk 2]
```diff
@@ -468,9 +468,14 @@ func adjustRelPath(hctx *hydrationContext) error {
 		if err != nil {
 			return err
 		}
+		// in kyaml v0.12.0, we are supporting both the new path annotation key
+		// internal.config.kubernetes.io/path, as well as the legacy one config.kubernetes.io/path
 		if err = r.PipeE(yaml.SetAnnotation(kioutil.PathAnnotation, newPath)); err != nil {
 			return err
 		}
+		if err = r.PipeE(yaml.SetAnnotation(kioutil.LegacyPathAnnotation, newPath)); err != nil {
```
[the remark this answers is not in the document]
Comment (droot) [reply internal/cmdrender/executor.go:476]: Aah. Interesting.

**Review by mortent — APPROVED** [review]

**Pull request merged**